File: 0206-reverse-linked-list/0206-reverse-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
    let curr = head;
    let prev = null;
    let next;
    
    while(curr != null){
        next = curr.next;    // curr.next node is saved in next
        curr.next = prev;    // current node is made to point previous
        prev = curr;         // prev node is now curr for next iteration
        curr = next;         // current node is made next for next iteration
    }
    return prev;
};

File: 0206-reverse-linked-list/0206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        #Iterative
        prev, current = None, head
        
        while current:
            nxt = current.next
            current.next = prev
            prev = current
            current = nxt
        return prev
        